src/test/java/bio/terra/app/usermetrics/BardClientTest.java

```diff
@@ -0,0 +1,169 @@
+package bio.terra.app.usermetrics;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.CALLS_REAL_METHODS;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.withSettings;
+
+import bio.terra.app.configuration.UserMetricsConfiguration;
+import bio.terra.common.category.Unit;
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import java.util.Map;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.web.client.RestTemplate;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(properties = {"datarepo.testWithEmbeddedDatabase=false"})
+@AutoConfigureMockMvc
+@ActiveProfiles({"google", "unittest"})
+@Category(Unit.class)
+@SuppressWarnings("rawtypes")
+public class BardClientTest {
+
+  @SpyBean UserMetricsConfiguration userMetricsConfiguration;
+
+  private AuthenticatedUserRequest user1;
+
+  private AuthenticatedUserRequest user2;
+
+  private String SYNC_PATH;
+
+  private String API_PATH;
+
+  @Before
+  public void setup() {
+    SYNC_PATH = userMetricsConfiguration.getBardBasePath() + "/api/syncProfile";
+    API_PATH = userMetricsConfiguration.getBardBasePath() + "/api/event";
+    user1 =
+        AuthenticatedUserRequest.builder()
+            .setSubjectId("Bob")
+            .setEmail("bob")
+            .setToken("bob token")
+            .build();
+
+    user2 =
+        AuthenticatedUserRequest.builder()
+            .setSubjectId("alice")
+            .setEmail("alice")
+            .setToken("alice token")
+            .build();
+  }
+
+  @Test
+  public void testBardClientLogEvent() {
+    BardClient bardClient =
+        mock(
+            BardClient.class,
+            withSettings()
+                .useConstructor(userMetricsConfiguration)
+                .defaultAnswer(CALLS_REAL_METHODS));
+    RestTemplate apiRestTemplate = mock(RestTemplate.class);
+    RestTemplate syncRestTemplate = mock(RestTemplate.class);
+    ResponseEntity responseEntity = mock(ResponseEntity.class);
+    when(responseEntity.getStatusCode()).thenReturn(HttpStatus.OK);
```

Comment:
In general I wouldn't mock a data object, only a service object, unless the data object has a private constructor or is otherwise awkward to construct. In this case there's a public constructor that should work
```suggestion
    ResponseEntity responseEntity = new ResponseEntity(HttpStatus.OK);
```